plugins/reach/lib/kReachManager.php

```diff
@@ -907,27 +911,33 @@ protected static function getEntryVendorTaskStatus($reachProfile, $vendorCatalog
 	private function checkAutomaticRules($object, $checkEmptyRulesOnly = false)
 	{
 		$scope = new kScope();
-		$vendorTaskObjectHandler = HandlerFactory::getHandlerByObject($object);
-		$taskObjectId = $vendorTaskObjectHandler->getTaskObjectId($object);
-		$scope->setEntryId($taskObjectId);
+		$scope->setEntryId($object->getEntryId());
 		$this->initReachProfileForPartner($object->getPartnerId());
 		if (self::$reachProfilesFilteredThatIncludesRegularRules)
 		{
 			foreach (self::$reachProfilesFilteredThatIncludesRegularRules as $profile)
 			{
 				/* @var $profile ReachProfile */
 				$fullFieldCatalogItemIds = $profile->fulfillsRules($scope, $checkEmptyRulesOnly);
-				if (!count($fullFieldCatalogItemIds))
-				{
-					continue;
-				}
-				$allowedCatalogItemIds = PartnerCatalogItemPeer::retrieveActiveCatalogItemIds($fullFieldCatalogItemIds, $object->getPartnerId());
-				if(!count($allowedCatalogItemIds))
+				foreach($fullFieldCatalogItemIds as $taskObjectType => $catalogItemIds)
 				{
-					KalturaLog::debug("None of the fulfilled catalog item ids are active on partner, [" . implode(",", $fullFieldCatalogItemIds) . "]");
-					continue;
+					if (!count($catalogItemIds))
+					{
+						continue;
+					}
+					$allowedCatalogItemIds = PartnerCatalogItemPeer::retrieveActiveCatalogItemIds($catalogItemIds, $object->getPartnerId());
```

Comment:
this is also a duplicate code and can go inside addingEntryVendorTaskByObjectIds